src/gameUtils.ts

```diff
@@ -51,7 +51,7 @@ type moveSnakeResp = {
 };
 export const moveSnake = (snake: Point[], direction: Point, food: Point | undefined): moveSnakeResp => {
   const newBody = [getPointOutOfGameBoard({ x: snake[0].x + direction.x, y: snake[0].y + direction.y }), ...snake];
-  const isEaten = typeof food !== 'undefined' && isFoodEaten(newBody, food);
+  const isEaten = Boolean(food && isFoodEaten(newBody, food));
```

Comment:
I think even better:

```suggestion
  const isEaten = food !== undefined && isFoodEaten(newBody, food));
```

This is a weird way to check for undefined:
```typescript
typeof food !== 'undefined'
```